src/main/java/org/jabref/logic/importer/util/GrobidService.java

```diff
@@ -0,0 +1,53 @@
+package org.jabref.logic.importer.util;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.jabref.logic.net.URLDownload;
+
+/**
+ * Implements an API to a GROBID server, as described at
+ * https://grobid.readthedocs.io/en/latest/Grobid-service/#grobid-web-services
+ * <p>
+ * Note: Currently a custom GROBID server is used...
+ * https://github.com/NikodemKch/grobid
+ * <p>
+ * The methods are structured to match the GROBID server api.
+ * Each method corresponds to a GROBID service request. Only the ones already used are already implemented.
+ */
+public class GrobidService {
+
+    public enum ConsolidateCitations {
+        NO(0), WITH_METADATA(1), WITH_DOI_ONLY(2);
+        private int code;
+
+        ConsolidateCitations(int code) {
+            this.code = code;
+        }
+
+        public int getCode() {
+            return this.code;
+        }
+    }
+
+    String grobidServerURL;
+
+    public GrobidService(String grobidServerURL) {
+        this.grobidServerURL = grobidServerURL;
+    }
+
+    public String processCitation(String rawCitation, ConsolidateCitations consolidateCitations) throws IOException {
+        rawCitation = URLEncoder.encode(rawCitation, StandardCharsets.UTF_8);
+        URLDownload urlDownload = new URLDownload(grobidServerURL
+                + "/api/processCitation");
+        urlDownload.setPostData("citations=" + rawCitation + "&consolidateCitations=" + consolidateCitations);
```

Comment:
As of now, for each reference (i.e. line) in the plaintext reference a new request to the grobid server is invoked. Is there really no batch processing where we can simply pass the complete text and GROBID returns a list of references (with only a single http request)?

Reply:
At least when it comes to http requests only these are available: https://grobid.readthedocs.io/en/latest/Grobid-service/#grobid-web-services

I already tried to send multiple entries but with no luck, they are not separated. (Thats why we do this on our own)

Maybe there is a possibility inside the GROBID library, but this would involve adding this big library via gradle. We already tried this but got a lot of errors because of the java modules thing which we believe was not our fault.

Another option would be to change the GROBID server a little bit more than we did now...

Reply:
Ok, then leave it as it is right now. Would be nice if you could open a feature request at GROBID since I guess this would be a usable additional for other projects as well.

Reply:
I have send a request via issue here: https://github.com/kermitt2/grobid/issues/540